### PR TITLE
Render url without images

### PR DESCRIPTION
**Pull request by qptr**

### CODE REVIEW

File: src/reader/handlers.clj [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+            [ring.util.response :as response]
+            [ring.middleware.params :refer [wrap-params]]))
+
+(defn fetch-page
+  [{:keys [query-params]}]
+  (if-let [url (get query-params "url")]
+    (response/response (html/html-without-images (html/fetch-html url)))
+    (response/bad-request "URL query missing")))
+
+(def handler (wrap-params #'fetch-page))
```
Comment (jysandy) [inline src/reader/handlers.clj:12]: This doesn't need to be a `var`.
```suggestion
(def handler (wrap-params fetch-page))
```

File: test/reader/handlers_test.clj [hunk 2]
```diff
@@ -0,0 +1,14 @@
+(ns reader.handlers_test
```
Comment (jysandy) [inline test/reader/handlers_test.clj]:
```suggestion
(ns reader.handlers-test
```

File: test/reader/html_test.clj [hunk 3]
```diff
@@ -0,0 +1,16 @@
+(ns reader.html_test
```
Comment (jysandy) [inline test/reader/html_test.clj]:
```suggestion
(ns reader.html-test
```

File: test/reader/handlers_test.clj [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+(deftest handler-test
+  (testing "Given url query parameter, return status 200 response"
+    (with-redefs [html/html-without-images (constantly "hello world")]
+      (let [request {:query-params {"url" "foo.bar"}}
+            response (h/fetch-page request)]
+        (is (= 200 (:status response))))))
+  (testing "Given request without url parameter, return status 400 response"
+    (let [response (h/fetch-page {})]
+      (is (= 400 (:status response))))))
```
Comment (jysandy) [inline test/reader/handlers_test.clj]: Add a newline at the end of each file. You can configure IntelliJ to do this for you.

File: test/reader/html_test.clj [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    (let [html "<html><head></head><body></body></html>"]
+      (is (= html (h/html-without-images html)))))
+  (testing "Given html with img/svg tags, return html without them"
+    (let [html1 "<html><head></head><body><img src='' alt=''></body></html>"
+          html2 "<html><head></head><body><svg width='' height=''></svg></body></html>"
+          html3 "<html><head></head><body><img src='' alt=''><svg></svg></body></html>"
+          changed-html "<html><head></head><body></body></html>"]
+      (is (= changed-html (h/html-without-images html1)))
+      (is (= changed-html (h/html-without-images html2)))
+      (is (= changed-html (h/html-without-images html3))))))
```
Comment (jysandy) [inline test/reader/html_test.clj]: I think it's worth having one more test case where the `img` tag is more deeply nested, perhaps under a few `div`s. Also, add a newline at the end of the file.

File: src/reader/html.clj [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  (:require [hickory.core :as h]
+            [hickory.render :as hr]
+            [clojure.walk :as w]
+            [org.httpkit.client :as c]))
+
+(defn fetch-html
+  [url]
+  (:body @(c/get url)))
+
+(defn html-to-hickory
```
Comment (jysandy) [inline src/reader/html.clj]: If a function is not used outside of its namespace, mark it as private.
```suggestion
(defn- html-to-hickory
```

File: src/reader/html.clj [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  [html]
+  (h/as-hickory (h/parse html)))
+
+(defn remove-node-images
```
Comment (jysandy) [inline src/reader/html.clj]:
```suggestion
(defn- remove-node-images
```

File: src/reader/html.clj [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  [node]
+  (if (and (map? node)
+           (or (= (:tag node) :img)
+               (= (:tag node) :svg)))
+    ""
+    node))
+
+(defn remove-images
```
Comment (jysandy) [inline src/reader/html.clj]:
```suggestion
(defn- remove-images
```

**Review by jysandy — CHANGES_REQUESTED** [review]
Overall, the code seems reasonably well-factored, and I don't have any comments beyond these minor changes.

File: src/reader/handlers.clj (hunk 1 above)
Comment (qptr) [reply src/reader/handlers.clj:12]: wait this is to avoid restarting the server?

Comment (jysandy) [reply src/reader/handlers.clj:12]: To avoid restarting the server, you'll have to pass a `var` to `run-jetty`.
Is this even solving the problem?

Comment (qptr) [reply src/reader/handlers.clj:12]: handler is passed to run-jetty. yeah it's been working?

Comment (qptr) [reply src/reader/handlers.clj:12]: ok it works without as well.

Comment (qptr) [reply src/reader/handlers.clj:12]: ok it works without as well (i'll re-read vars)

**Pull request merged**